### PR TITLE
fix(NavigationManager): fix render order and storybook glitches

### PR DESCRIPTION
**Pull request by ChrisArasin**
## Description

- Allows NavigationManager children components to run layout/updates before the parent does. This prevents a mis-rendered stack of children on the first render of column/row.
- Also includes Storybook render glitch fixes to normalize storybook rendering. These changes are described here: https://github.com/rdkcentral/Lightning-UI-Components/pull/465
- It helps to evaluate this PR, but we can merge separately or all together here. 
- Fixes the use of the centerKeys prop in keyboard to fix a mis-rendered first draw. 

## References

Related to: [LUI-1328](https://ccp.sys.comcast.net/browse/LUI-1328)

## Testing

- Ensure Column/Row based layouts are functioning properly. Items should draw in place and not stacked on top of each other when chrome is throttled down.

## Checklist

- [x] all commented code has been removed
- [x] any new console issues have been resolved
- [x] code linter and formatter has been run
- [x] test coverage meets repo requirements
- [x] PR name matches the expected semantic-commit syntax

### CODE REVIEW

File: packages/@lightningjs/ui-components/src/components/Row/Row.test.js [hunk 1]
```diff
@@ -431,31 +431,31 @@ describe('Row', () => {
       done();
     });
 
-    it('should pass on screen items to onScreenEffect', done => {
+    it('should pass on screen items to onScreenEffect', async () => {
       row.w = 200;
+      row.clipping = true;
```
Comment (ChrisArasin) [inline packages/@lightningjs/ui-components/src/components/Row/Row.test.js:436]: This test needs clipping to pass as expected. I'm not sure why it was passing before...

File: packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.js [hunk 2]
```diff
@@ -114,6 +113,9 @@ export default class FocusManager extends Base {
       x: this.itemPosX,
       y: this.itemPosY
     });
+    if (this._lazyItems) {
+      this._lazyItems = null;
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.js:120]: does it matter that this is null instead of an empty array

Comment (ChrisArasin) [reply packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.js:120]: Yeah, this is actually broken on develop now too. NavigationManager's `appendItems()` method does this check, so if that array exists when set items is called they just get pushed there and not rendered.

```
  appendItems(items = []) {
    this.shouldSmooth = false;

    if (this._lazyItems) {
      this._lazyItems.push(...items);
      return;
    }
 ```

File: packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.d.ts [hunk 3]
```diff
@@ -176,8 +176,9 @@ declare class FocusManager<
   /**
    * Selects next item. If this.wrapSelected=true, will select the first element in the list if focus is currently on the last item.
    * emits the `selectedChange` signal
+   * @param forceSmoothValue allows setting the shouldSmooth value before the select update
    */
-  selectNext(): void;
+  selectNext(forceSmoothValue?: boolean): void;
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.d.ts]: mind updating FocusManager.mdx too? Line 162 has selectNext

Comment (ChrisArasin) [reply packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.d.ts]: yep, gave it a better name too

**Comment by ImCoolNowRight** [comment]
All the stories I tested are looking good, but I did see that the TextPanel portion of GridOverlay still has a Column that is showing the TextBoxes stacked when first rendering:


https://github.com/rdkcentral/Lightning-UI-Components/assets/2678766/f5974b6d-b632-461a-9c15-af839961ef04

**Comment by ChrisArasin** [comment]
> All the stories I tested are looking good, but I did see that the TextPanel portion of GridOverlay still has a Column that is showing the TextBoxes stacked when first rendering:

So this would be an example of where waitForDimensions may still be relevant. In this case, the items don't have a height until the text textures are rendered and the textboxes update. If they had a height set beforehand, then would not stack

**Review by ImCoolNowRight — APPROVED** [review]

File: packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.js (hunk 2 above)
Comment (erautenberg) [reply packages/@lightningjs/ui-components/src/components/FocusManager/FocusManager.js:120]: *will handle error checking for this in a separate ticket

**Review by erautenberg — APPROVED** [review]

**Pull request merged**